pkg/crypto/fingerprint/hash.go

```diff
@@ -16,10 +16,26 @@ var nameToHash = map[string]crypto.Hash{
 	"sha-512": crypto.SHA512, // [RFC4055]
 }
 
+var hashToName = make(map[crypto.Hash]string)
+
+func init() {
```

Comment:
gochecknoinits

Reply:
I think it is fine to run this every time! I would just have `StringToHash` be more expensive.